### PR TITLE
Api albums

### PR DESCRIPTION
**Pull request by YesicaNavaB**
## Summary

Integrations Api Albums

## Trello Card

https://trello.com/c/bWWJws0H

### CODE REVIEW

File: app/controllers/albums.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable func-names */
+const servicesAlbums = require('../services/albums');
+
+exports.getAlbums = (req, res, next) =>
+  servicesAlbums
+    .getAlbumsApi()
+    .then(albums => res.send({ albums }))
+    .catch(() => next);
```
Comment (jersonlopez) [inline app/controllers/albums.js]: `.catch(next);` is enough

File: app/controllers/albums.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+exports.getAlbumsPhotos = function(req, res, next) {
+  const idAlbum = req.params.id;
+  servicesAlbums
+    .getAlbumsPhotosApi(idAlbum)
+    .then(albums => res.send({ albums }))
+    .catch(() => next);
```
Comment (jersonlopez) [inline app/controllers/albums.js]: same here!

File: app/services/albums.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const request = require('request-promise');
+
+exports.getAlbumsApi = () =>
+  request('https://jsonplaceholder.typicode.com/albums')
+    .then(htmlString => JSON.parse(htmlString))
+    .catch(error => {
+      throw error;
```
Comment (jersonlopez) [inline app/services/albums.js]: the error must be logged, in Wolox we have a customize logger that you can found in app/logger. In this case is something like `logger.error(myError)`.


If you have any questions please let me know and we try to solve it together

File: app/services/albums.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const request = require('request-promise');
+
+exports.getAlbumsApi = () =>
+  request('https://jsonplaceholder.typicode.com/albums')
+    .then(htmlString => JSON.parse(htmlString))
```
Comment (jersonlopez) [inline app/services/albums.js]: `JSON.parse(htmlString)` is really necessary?

File: app/services/albums.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const request = require('request-promise');
+
+exports.getAlbumsApi = () =>
+  request('https://jsonplaceholder.typicode.com/albums')
```
Comment (jersonlopez) [inline app/services/albums.js]: here are several comments:
- what happen if the user wants a specific album?
- if the album's url endpoint changes, you will have to come to the code and change it in each occurrence i think that it must be an environment variable that you can expose by config files.

File: app/services/albums.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    .then(htmlString => JSON.parse(htmlString))
+    .catch(error => {
+      throw error;
+    });
+
+exports.getAlbumsPhotosApi = idAlbum =>
```
Comment (jersonlopez) [inline app/services/albums.js]: the same comments from behind, and think about making a more generic method

**Review by jersonlopez — CHANGES_REQUESTED** [review]

File: app/services/albums.js (hunk 3 above)
Comment (YesicaNavaB) [reply app/services/albums.js]: Perfect!

File: app/controllers/albums.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+const servicesAlbums = require('../services/albums'),
+  logger = require('../logger'),
+  config = require('../../config'),
+  { url } = config.common.ApiAlbums;
+let source = url;
+
+exports.getAlbums = (req, res, next) => {
+  const idAlbum = req.params.id;
+  logger.info(`${url}/albums/${idAlbum}`);
+  source = idAlbum === undefined ? `${url}/albums` : `${url}/albums/${idAlbum}`;
```
Comment (jersonlopez) [inline app/controllers/albums.js]: ```source = !idAlbum ? `${url}/albums` : `${url}/albums/${idAlbum}`;```

File: app/controllers/albums.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+const servicesAlbums = require('../services/albums'),
+  logger = require('../logger'),
+  config = require('../../config'),
+  { url } = config.common.ApiAlbums;
+let source = url;
+
+exports.getAlbums = (req, res, next) => {
+  const idAlbum = req.params.id;
+  logger.info(`${url}/albums/${idAlbum}`);
```
Comment (jersonlopez) [inline app/controllers/albums.js]: unnecessary log

File: app/services/albums.js (hunk 5 above)
Comment (YesicaNavaB) [reply app/services/albums.js]: Thanks for the comment.

File: app/services/albums.js (hunk 6 above)
Comment (YesicaNavaB) [reply app/services/albums.js]: Perfect!

File: app/services/albums.js (hunk 4 above)
Comment (YesicaNavaB) [reply app/services/albums.js]: not is necessary!

File: app/controllers/albums.js (hunk 7 above)
Comment (YesicaNavaB) [reply app/controllers/albums.js]: source = idAlbum ?  `${url}/albums/${idAlbum}` : `${url}/albums`;

File: app/controllers/albums.js (hunk 8 above)
Comment (YesicaNavaB) [reply app/controllers/albums.js]: ok

**Review by jersonlopez — APPROVED** [review]
good job!! 🥇 👏

**Pull request merged**